collect_app/src/main/java/org/odk/collect/android/adapters/FileArrayAdapter.java

```diff
@@ -103,4 +115,15 @@ public View getView(int position, View convertView, @NonNull ViewGroup parent) {
         holder.onBind(items.get(position));
         return view;
     }
+
+    private boolean isNewVersion(DriveListItem item) {
+        FormsDao formsDao = new FormsDao();
+        Cursor cursor = formsDao.getFormsCursorForFormFilePath(Collect.FORMS_PATH + File.separator + item.getName());
+        if (cursor != null && cursor.getCount() > 0) {
+            Long lastModifiedLocal = new File(Collect.FORMS_PATH + File.separator + item.getName()).lastModified();
```

Comment:
If you already have cursor it's better to read lastModifiedLocal from its DATE column.

Reply:
I tried using that but that poses a problem. Since in this case we do not have access to md5 or any other form specific data it is possible that the form available on google drive is essentially the same form(not an update!) but it would show as an update to the user because we're comparing modified time on gdrive and so if the user chooses to download that form, the database doesn't update(because we have md5 check there) so the DATE column also doesn't update, so the user will still see that the update is available.

Reply:
@grzesiek2010 can we ever face the problem i mentioned above or we're fine using DATE column??

Reply:
@chidauri this would happen if a user did something like change the form definition and then change their mind and change it back, right? That seems like an uncommon enough scenario that I wouldn’t worry about it. Is there another case that would be affected?

There is possible weirdness if the user’s device has the wrong date/time, around daylight savings boundaries and if the device is in a time zone other than the one Google Dive is in but I think these are ok to document and not worry about.

Reply:
@lognaturel this would also happen if we upload the form we already have to google drive. Also, is it fine to use last modified time for media files too?

Reply:
Oh I see thanks for the explanation. In this case leaving this as is seems fine to me.